blob/helper.go

```diff
@@ -0,0 +1,52 @@
+package blob
+
+import (
+	"github.com/celestiaorg/celestia-app/pkg/appconsts"
+	"github.com/celestiaorg/celestia-app/pkg/shares"
+)
+
+// sharesToBlobs takes raw shares and converts them to blobs.
+func sharesToBlobs(rawShares [][]byte) ([]*Blob, error) {
+	if len(rawShares) == 0 {
+		return nil, ErrBlobNotFound
+	}
+
+	shareSequences, err := shares.ParseShares(rawShares)
+	if err != nil {
+		return nil, err
+	}
+
+	blobs := make([]*Blob, len(shareSequences))
+	for i, sequence := range shareSequences {
+		data, err := sequence.RawData()
+		if err != nil {
+			return nil, err
+		}
+		if len(data) == 0 {
+			continue
+		}
+
+		blob, err := NewBlob(data[0], sequence.NamespaceID, data)
+		if err != nil {
+			return nil, err
+		}
+		blobs[i] = blob
+	}
+	return blobs, nil
+}
+
+const (
+	perByteGasTolerance = 2
+	pfbGasFixedCost     = 80000
+)
+
+// estimateGas estimates the gas required to pay for a set of blobs in a PFB.
+func estimateGas(blobs ...*Blob) uint64 {
+	totalByteCount := 0
+	for _, blob := range blobs {
+		totalByteCount += len(blob.Data()) + appconsts.NamespaceSize
+	}
+	variableGasAmount := (appconsts.DefaultGasPerBlobByte + perByteGasTolerance) * totalByteCount
```

Comment:
So the only problematic thing here is that `appconsts.DefaultGasPerBlobByte` is the default cost but it can be modified through governance i.e. Celestia could have a proposal to bump the gas per blob byte to 20 instead of 8 and then this gas estimation would be irrelevant. cc @evan-forbes

Reply:
also, if this is just being used to determine the gas limit, then I don't think we have to be as precise and would want to error on the side of over estimation as there are many other things can cause the gas of tx to increase.

Reply:
Alternatively we could use https://docs.cosmos.network/swagger/#/Service/Simulate to always query for the expected gas amount and then submit the transaction with 1.2x

Reply:
I do think it would be an important development to have gas refunds. This would allow people to not worry about having a good estimation

Reply:
So, should I rework the gas estimation? or we can leave it as it is now

Reply:
I might change it to calculate per share instead of per byte if we want to be super accurate, but can defo be done later if it actually becomes an issue. Also, basing off a configured value instead of a constant might be a good provided we add enough documentation to let users know that it likely doesn't need to be changed unless they're getting errors for running out of gas.